src/local/exm-manager.c

```diff
@@ -353,52 +353,42 @@ exm_manager_is_installed_uuid (ExmManager  *self,
     return FALSE;
 }
 
-static void
-do_install_thread (GTask        *task,
-                   ExmManager   *self,
-                   const char   *uuid,
-                   GCancellable *cancellable)
-{
-    GError *error = NULL;
-
-    g_dbus_proxy_call_sync (G_DBUS_PROXY (self->proxy),
-                            "InstallRemoteExtension",
-                            g_variant_new ("(s)", uuid, NULL),
-                            G_DBUS_CALL_FLAGS_NONE,
-                            -1, cancellable, &error);
-
-    if (error != NULL)
-    {
-        g_task_return_error (task, error);
-        return;
-    }
-
-    g_task_return_boolean (task, TRUE);
-}
-
 void
 exm_manager_install_async (ExmManager          *self,
                            const gchar         *uuid,
                            GCancellable        *cancellable,
                            GAsyncReadyCallback  callback,
                            gpointer             user_data)
 {
-    GTask *task;
-
-    task = g_task_new (self, cancellable, callback, user_data);
-    g_task_set_task_data (task, g_strdup (uuid), (GDestroyNotify) g_free);
-    g_task_run_in_thread (task, (GTaskThreadFunc)do_install_thread);
-    g_object_unref (task);
+    shell_extensions_call_install_remote_extension (self->proxy,
+                                                    uuid,
+                                                    cancellable,
+                                                    callback,
+                                                    user_data);
```

Comment:
I don't know if there was any reason originally for not using `shell_extensions_call_install_remote_extension` directly.